ngraph/frontend/onnx_import/include/onnx_import/editor/editor.hpp

```diff
@@ -70,25 +71,47 @@ namespace ngraph
             ///                     the inputs specified in its parameter.
             void set_input_shapes(const std::map<std::string, ngraph::PartialShape>& input_shapes);
 
+            /// \brief Extracts a subgraph constrained by input edges and output edges. In the end
+            ///        the underlying ModelProto is modified - obsolete inputs, initializers, nodes
+            ///        and outputs are removed from the in-memory model.
+            ///
+            /// \node Please look at the declaration of InputEdge and OutputEdge for explanation
+            ///       how those objects can be created. If the outputs parameter is empty
+            ///       this method keeps all of the original outputs of the model.
+            ///
+            /// \param inputs A collection of input edges which become new inputs to the graph
+            /// \param outputs A collection of output edges which become new outputs of the graph
+            void cut_graph_fragment(const std::vector<InputEdge>& inputs,
+                                    const std::vector<OutputEdge>& outputs);
+
             /// \brief Returns a non-const reference to the underlying ModelProto object, possibly
             ///        modified by the editor's API calls
             ///
             /// \return A reference to ONNX ModelProto object containing the in-memory model
             ONNX_NAMESPACE::ModelProto& model() const;
 
+            /// \brief Returns a serialized ONNX model, possiblie modified by an instance of editor.
+            std::string model_string() const;
```

Comment:
Does it means that we can remove import_network method for `ONNXEditor`?

Reply:
Perhaps we should go this way but let me discuss it with the team today and get back to you.

Reply:
This will be benchmarked and addressed in a separate PR where the editor will be extracted to a standalone library.